BadHabbit/BadHabbit/UserStories/SingUpScreen/SignUpViewController.swift

```diff
@@ -0,0 +1,173 @@
+        welcomeLabel.textAlignment = .center
+        welcomeLabel.textColor = .black
+    }
+    
+    private func addAppNameLabel() {
+        view.addSubview(appNameLabel)
+
+        appNameLabel.text = String(localized: "appName").uppercased()
+        appNameLabel.font = UIFont(name: FontName.SFProDisplayBold.rawValue, size: 24.0)
+        appNameLabel.textAlignment = .center
+        appNameLabel.textColor = .black
+    }
+    
+    private func addLogoImageView() {
+        view.addSubview(logoImageView)
+        addLogoLabel()
+        
+        logoImageView.image = UIImage(named: "logo")
+    }
+    
+    private func addLogoLabel() {
+        logoImageView.addSubview(logoLabel)
+        
+        logoLabel.text = String(localized: "signUpScreen.logoLabelText").uppercased()
+        logoLabel.font = UIFont(name: FontName.interExtraBold.rawValue, size: 15.0)
+        logoLabel.textAlignment = .left
+        logoLabel.textColor = UIColor(hexNumber: 0xFFFFFF)
+    }
+    
+    private func addPhraseLabel() {
+        view.addSubview(phraseLabel)
+        
+        phraseLabel.text = String(localized: "signUpScreen.phraseLabelText").uppercased()
+        phraseLabel.font = UIFont(name: FontName.SFProDisplayMedium.rawValue, size: 14.0)
+        phraseLabel.textAlignment = .center
+        phraseLabel.textColor = .black
+        phraseLabel.lineBreakMode = .byWordWrapping
+        phraseLabel.numberOfLines = 0
+    }
+    
+    private func addStartButton() {
+        view.addSubview(startButton)
+        
+        var buttonConfiguration = UIButton.Configuration.filled()
+        buttonConfiguration.title = String(localized: "signUpScreen.startButtonTitle").uppercased()
+        buttonConfiguration.cornerStyle = .large
+        buttonConfiguration.baseBackgroundColor = UIColor(hexNumber: 0x090909)
+
+        startButton.configuration = buttonConfiguration
+        
+        startButton.addTarget(self, action: #selector(startButtonTapped), for: .touchUpInside)
+    }
+    
+    private func setupViewConstraints() {
+        let safeArea = view.safeAreaLayoutGuide
+        
+        backgroundImageView.snp.makeConstraints { make in
+            let heightMultiplier: CGFloat = 445 / 844
+
+            make.height.equalToSuperview().multipliedBy(heightMultiplier)
+            make.bottom.equalToSuperview()
+            make.leading.trailing.equalTo(safeArea)
+        }
+        
+        welcomeLabel.snp.makeConstraints { make in
+            make.top.equalTo(safeArea).inset(35)
+            make.centerX.equalTo(safeArea)
+        }
+        
+        appNameLabel.snp.makeConstraints { make in
+            make.top.equalTo(welcomeLabel.snp.bottom)
+            make.centerX.equalTo(safeArea)
+        }
+        
+        safeArea.snp.makeConstraints { make in
+            make.centerY.equalTo(logoImageView).offset(35)
+        }
+        
+        logoImageView.snp.makeConstraints { make in
+            let multiplayer: CGFloat = 196 / 390
```

Comment:
It's bad practices to write some numbers which is clear only for you. The called "magic numbers"

https://www.inf.unibz.it/~calvanese/teaching/04-05-ip/lecture-notes/uni04/node15.html#:~:text=A%20magic%20number%20is%20a,difficult%20to%20maintain%20and%20update.

I suppose it's width and height of image. If possible - get this values out of image

Reply:
I aded a sizeMultiplier constant and assigned it a closure. Is it possible?